### ISSUE TITLE
Diffusion model generalization ability

### ISSUE THREAD

**Issue by 18hfliu** [BioinfoMachineLearning/bio-diffusion#5]
When I try to train DEM and GeoLDM with my own dataset, gradient norm is too large. an the gradient norm decrease when training a bio-diffusion model on my dataset? My custom data set is different from qm9, the molecule contains up to 74 atoms. Atomic categories are 'H', 'C', 'N', 'O', 'F', 'P', 'S', 'Cl', 'Br', a total of 9 categories.

**Comment by amorehead** [comment]
Hello. Based on my experience training GCDM (Bio-Diffusion) models on QM9 and GEOM-Drugs, the gradient norm typically begins to stabilize after the first few training epochs, at which point the model begins to generate increasingly more stable and valid molecules. By default, `clip_gradients` is set to `True` (e.g., https://github.com/BioinfoMachineLearning/bio-diffusion/blob/775a9760972f6ec254c44dd6e623c4451403ca4f/src/models/qm9_mol_gen_ddpm.py#L147) such that the model's gradients are clipped according to a buffer of recent gradient values (ref: https://github.com/BioinfoMachineLearning/bio-diffusion/blob/775a9760972f6ec254c44dd6e623c4451403ca4f/src/models/qm9_mol_gen_ddpm.py#L1263). In practice, this means that once the model has spent the first few epochs making small adjustments to the initial layers' weights, the network begins to have increasingly stable gradients. Assuming this custom dataset you mentioned is relatively clean (e.g., doesn't contain excessive noise), I believe this trend should hold.

Moreover, we notice improved model stability over larger datasets such as GEOM-Drugs, which consequently allows GCDM to generate more stable large molecules. Whether your custom dataset's molecules are large or small, you should be able to fit a model over these molecules assuming they aren't excessively noisy.

**Comment by 18hfliu** [comment]
I try to denoise the dataset (e.g., use DFT to generate more accurate 3D coordinates). Can a GCDM (biodiffusion) model be trained using the GEOM-Drugs dataset by condition and then generate new conditional 3D molecules?

**Comment by amorehead** [comment]
Hello. Yes, in principle, if you have a custom dataset (similar to GEOM-Drugs) that contains property values for each 3D molecule, GCDM should be able to accurately model the conditional distribution of molecular property values, even for larger GEOM-Drugs-sized molecules (since GCDM seems to perform markedly better than previous baselines such as EDM and GeoLDM for unconditional GEOM-Drugs molecule generation).

Related to this point, I will soon be updating GCDM's arXiv preprint with expanded results for QM9 and GEOM-Drugs, further illustrating where GCDM shines for large 3D molecule generation (such as with GEOM-Drugs).

**Comment by amorehead** [comment]
If you have any other questions, please let me know. For the time being, I will be closing this issue.
